docs/apis/calcite.md

```diff
@@ -0,0 +1,235 @@
+|**SQLExecEnvironment.registerFunction(sqlFunctionName, holderClass, staticFunctionName)**| Registers a new User Defined Scalar function |
+|**SQLExecEnvironment.executeSQL(dag, sqlStatement)** | Creates a DAG for a particular SQL statement |
+
+Usage of above APIs is described in details in following sections.
+
+## Example 1: Pure Style SQL Application
+
+With Apache Calcite Integration, you can use SQL queries across different data sources and provide UDFs as per your business logic. This example will use Kafka as input source and HDFS file as the output destination.
+Following application code will be used to explain APIs. Actual source code can be found [here](https://github.com/apache/apex-malhar/blob/master/demos/sql/src/main/java/org/apache/apex/malhar/sql/sample/PureStyleSQLApplication.java).
+```java
+public class PureStyleSQLApplication implements StreamingApplication
+{
+ @Override
+ public void populateDAG(DAG dag, Configuration conf)
+ {
+   // Create new SQLExecEnvironment
+   SQLExecEnvironment sqlEnv = SQLExecEnvironment.getEnvironment();
+   
+   // Register KafkaEnpoint as "ORDERS" table with kafka topic and data format as CSV
+   sqlEnv = sqlEnv.registerTable( 
+                                  "ORDERS", 
+                                  new KafkaEndpoint("localhost:9090", 
+                                                    "inputTopic", 
+                                                    new CSVMessageFormat(<>))
+                                );
+   
+   // Register FileEndpoint as "SALES" table with file path and data format as csv
+   sqlEnv = sqlEnv.registerTable( 
+                                  "SALES", 
+                                  new FileEndpoint("/tmp/output", 
+                                                   "out.file", 
+                                                   new CSVMessageFormat(<>))
+                                );
+                                
+   // Register scalar SQL UDF 
+   sqlEnv = sqlEnv.registerFunction("APEXCONCAT", PureStyleSQLApplication.class, "apex_concat_str");
+   
+   // Converting SQL statement to DAG 
+   String sql = "INSERT INTO SALES 
+                     SELECT STREAM ROWTIME, FLOOR(ROWTIME TO DAY), APEXCONCAT('OILPAINT', SUBSTRING(PRODUCT, 6, 7)) 
+                     FROM ORDERS 
+                     WHERE ID > 3 AND PRODUCT LIKE 'paint%'";
+   sqlEnv.executeSQL(dag, sql);
+ }// populateDAG finished
+ 
+ public static String apex_concat_str(String s1, String s2)
+ {
+      return s1 + s2;
+ } 
+}
+```      
+### Constructing SQLExecEnvironment
+Class _**SQLExecEnvironment**_ provides a starting point and a simple way to define metadata needed for running SQL statement. SQLExecEnvironment.getEnvironment() creates a new instance of SQLExecEnvironment.  
```

Comment:
Class => The class
"running SQL statement. SQLExecEnvironment.getEnvironment() creates a new instance of SQLExecEnvironment" => running an SQL statement; a new instance of this class is returned by the `getEnvironment` static method.